c2corg_api/views/sitemap.py

```diff
@@ -0,0 +1,148 @@
+import logging
+
+from c2corg_api import DBSession
+from c2corg_api.models.cache_version import CacheVersion
+from c2corg_api.models.document import Document, DocumentLocale
+from c2corg_api.models.route import ROUTE_TYPE, RouteLocale
+from c2corg_api.models.user_profile import USERPROFILE_TYPE
+from c2corg_api.views import cors_policy
+from c2corg_api.views.validation import create_int_validator, \
+    validate_document_type
+from cornice.resource import resource, view
+from pyramid.httpexceptions import HTTPNotFound
+from sqlalchemy.sql.functions import func
+from math import ceil
+
+log = logging.getLogger(__name__)
+
+# Search engines accept not more than 50000 urls per sitemap,
+# see http://www.sitemaps.org/protocol.html
+PAGES_PER_SITEMAP = 50000
+
+
+validate_page = create_int_validator('i')
+
+
+@resource(
+    collection_path='/sitemaps', path='/sitemaps/{doc_type}/{i}',
+    cors_policy=cors_policy)
+class SitemapRest(object):
+
+    def __init__(self, request):
+        self.request = request
+
+    @view()
+    def collection_get(self):
+        """ Returns the information needed to generate a sitemap index file.
+        See: http://www.sitemaps.org/protocol.html
+
+        The response consists of a list of URLs to request the information
+        needed to generate the sitemap linked from the sitemap index.
+
+        E.g.
+
+            {
+                "sitemaps": [
+                    "/sitemaps/w/0",
+                    "/sitemaps/a/0",
+                    "/sitemaps/i/0",
+                    "/sitemaps/i/1",
+                    "/sitemaps/i/2",
+                    "/sitemaps/i/3",
+                    "/sitemaps/i/4",
+                    "/sitemaps/i/5",
+                    ...
+                ]
+            }
+        """
+        document_locales_per_type = DBSession. \
+            query(Document.type, func.count().label('count')). \
+            join(
+                DocumentLocale,
+                Document.document_id == DocumentLocale.document_id). \
+            filter(Document.type != USERPROFILE_TYPE). \
```

Comment:
Does it mean that the user profile are excluded from the sitemaps? If yes that looks good to me.